e2e/restore_test.go

```diff
@@ -291,3 +301,431 @@ func TestSingleNodeAirgapDisasterRecovery(t *testing.T) {
 
 	t.Logf("%s: test complete", time.Now().Format(time.RFC3339))
 }
+
+func TestMultiNodeHADisasterRecovery(t *testing.T) {
+	t.Parallel()
+
+	requiredEnvVars := []string{
+		"DR_AWS_S3_ENDPOINT",
+		"DR_AWS_S3_REGION",
+		"DR_AWS_S3_BUCKET",
+		"DR_AWS_S3_PREFIX",
+		"DR_AWS_ACCESS_KEY_ID",
+		"DR_AWS_SECRET_ACCESS_KEY",
+	}
+	for _, envVar := range requiredEnvVars {
+		if os.Getenv(envVar) == "" {
+			t.Fatalf("missing required environment variable: %s", envVar)
+		}
+	}
+
+	testArgs := []string{}
+	for _, envVar := range requiredEnvVars {
+		testArgs = append(testArgs, os.Getenv(envVar))
+	}
+
+	tc := cluster.NewTestCluster(&cluster.Input{
+		T:                   t,
+		Nodes:               3,
+		Image:               "debian/12",
+		LicensePath:         "snapshot-license.yaml",
+		EmbeddedClusterPath: "../output/bin/embedded-cluster",
+	})
+	defer cleanupCluster(t, tc)
+
+	// install "expect" dependency on node 0 as that's where the restore process will be initiated.
```

Comment:
Should we run the restore from something other than node 0?

Just to make sure it works on a different node?

Reply:
I tried that, but hit a bunch of issues and had to give up given the time. It's because playwright is installed on node 0 by default and uses the node 0 IP as base URL. Doable, I don't wanna waste another day battling it though.